desktop/core/src/desktop/lib/raz/raz_client.py

```diff
@@ -100,47 +111,62 @@ def check_access(self, method, url, params=None, headers=None):
-    headers = [raz_signer.StringStringMapProto(key=key, value=val) for key, val in headers.items()]
-    endpoint = "%s://%s" % (path.scheme, path.netloc)
-    resource_path = path.path.lstrip("/")
-
-    LOG.debug(
-      "Preparing sign request with http_method: {%s}, headers: {%s}, parameters: {%s}, endpoint: {%s}, resource_path: {%s}" %
-      (method, headers, allparams, endpoint, resource_path)
-    )
-    raz_req = raz_signer.SignRequestProto(
-        endpoint_prefix=self.service_params['endpoint_prefix'],
-        service_name=self.service_params['service_name'],
-        endpoint=endpoint,
-        http_method=method,
-        headers=headers,
-        parameters=allparams,
-        resource_path=resource_path,
-        time_offset=0
-    )
-    raz_req_serialized = raz_req.SerializeToString()
-    signed_request = base64.b64encode(raz_req_serialized)
-
     request_data = {
       "requestId": self.requestid,
       "serviceType": self.service_params['serviceType'],
       "serviceName": self.service_name,
       "user": self.username,
       "userGroups": [],
-      "accessTime": "",
-      "clientIpAddress": "",
-      "clientType": "",
+      "clientIpAddress": self.client_params['clientIpAddress'],
+      "clientType": self.client_params['clientType'],
       "clusterName": self.cluster_name,
       "clusterType": "",
-      "sessionId": "",
-      "context": {
-        "S3_SIGN_REQUEST": signed_request
-      }
+      "context": {}
     }
-    headers = {"Content-Type":"application/json", "Accept-Encoding":"gzip,deflate"}
-    raz_url = "%s/api/authz/s3/access?delegation=%s" % (self.raz_url, self.raz_token)
+
+    if self.service == 's3':
+      allparams = [raz_signer.StringListStringMapProto(key=key, value=[val]) for key, val in url_params.items()]
+      allparams.extend([raz_signer.StringListStringMapProto(key=key, value=[val]) for key, val in params.items()])
+      headers = [raz_signer.StringStringMapProto(key=key, value=val) for key, val in headers.items()]
+      endpoint = "%s://%s" % (path.scheme, path.netloc)
+      resource_path = path.path.lstrip("/")
+
+      LOG.debug(
+        "Preparing sign request with http_method: {%s}, headers: {%s}, parameters: {%s}, endpoint: {%s}, resource_path: {%s}" %
+        (method, headers, allparams, endpoint, resource_path)
+      )
+      raz_req = raz_signer.SignRequestProto(
+          endpoint_prefix=self.service_params['endpoint_prefix'],
+          service_name=self.service_params['service_name'],
+          endpoint=endpoint,
+          http_method=method,
+          headers=headers,
+          parameters=allparams,
+          resource_path=resource_path,
+          time_offset=0
+      )
+      raz_req_serialized = raz_req.SerializeToString()
+      signed_request = base64.b64encode(raz_req_serialized)
+
+      request_data.update({
+        "sessionId": "",
+        "accessTime": "",
+        "context": {
```

Comment:
Here I was expecting ways simpler/shorter by not duplicating:

e.g.:

a parameterization of raz_url
request_data['context']= ... or request_data['operation'] = ...

operation content filled-up here too
-

Reply:
Populated the `request_data['operation']` with parameters from the raz_url and removed some duplication @romainr